exporter/datadogexporter/translate_traces.go

```diff
@@ -292,7 +292,7 @@ func spanToDatadogSpan(s pdata.Span,
 	span := &pb.Span{
 		TraceID:  decodeAPMTraceID(s.TraceID().Bytes()),
 		SpanID:   decodeAPMSpanID(s.SpanID().Bytes()),
-		Name:     remapDatadogSpanName(getDatadogSpanName(s, tags), spanNameMap),
+		Name:     remapDatadogSpanName(getDatadogSpanName(s, tags, cfg.Traces.SpanNameAsResourceName), spanNameMap),
```

Comment:
I really don't like boolean arguments to a function. Can we instead just do this:

```go
name := s.Name()
if !cfg.Traces.SpanNameAsResourceName {
    name = remapDatadogSpanName(getDatadogSpanName(s, tags), spanNameMap)
}
```
And then use `name`

Reply:
Thank you for your review! I removed the boolean parameter but I passed the name to the `remapDatadogSpanName` function, so the name can also be remapped after being set to the span name. See if that makes sense to you @gbbr!